src/CommonLib/Ntlm/LdapNative.cs

```diff
@@ -0,0 +1,346 @@
+            LdapOption.ResultCode,
+            out resultCode
+        );
+
+        CheckError(result, "Getting LDAP result code");
+        return resultCode;
+    }
+
+    public byte[] SaslBind(string distinguishedName, string mechanism, byte[] credential)
+    {
+        ThrowIfDisposed();
+
+        IntPtr credPtr = IntPtr.Zero;
+        try
+        {
+            // Create berval structure for credentials
+            var berval = new BERVAL
+            {
+                bv_len = credential.Length,
+                bv_val = Marshal.AllocHGlobal(credential.Length)
+            };
+
+            Marshal.Copy(credential, 0, berval.bv_val, credential.Length);
+
+            credPtr = Marshal.AllocHGlobal(Marshal.SizeOf<BERVAL>());
+            Marshal.StructureToPtr(berval, credPtr, false);
+
+            // Perform SASL bind
+            int result = NativeMethods.ldap_sasl_bind_s(
+                Handle,
+                distinguishedName,
+                mechanism,
+                credPtr,
+                IntPtr.Zero,  // No server controls
+                IntPtr.Zero,  // No client controls
+                out IntPtr response
+            );
+
+            if (result != (int)LdapErrorCodes.Success &&
+                result != (int)LdapErrorCodes.SaslBindInProgress)
+            {
+                throw new LdapException("SASL bind failed", result, GetServerError());
+            }
+
+            int bindResultCode = GetResultCode();
+
+            if (bindResultCode != (int)LdapErrorCodes.Success &&
+                bindResultCode != (int)LdapErrorCodes.SaslBindInProgress)
+            {
+                throw new LdapException("SASL bind result failed", bindResultCode, GetServerError());
+            }
+
+            if (response == IntPtr.Zero)
+                throw new InvalidOperationException("Server did return a challenge");
+
+            var responseBerval = Marshal.PtrToStructure<BERVAL>(response);
+            byte[] responseData = new byte[responseBerval.bv_len];
+            Marshal.Copy(responseBerval.bv_val, responseData, 0, responseBerval.bv_len);
+            return responseData;
+        }
+        finally
+        {
+            if (credPtr != IntPtr.Zero)
+            {
+                var berval = Marshal.PtrToStructure<BERVAL>(credPtr);
+                if (berval.bv_val != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(berval.bv_val);
+                }
+                Marshal.FreeHGlobal(credPtr);
+            }
+
+            // Freeing the memory always results in crashes. Is it managed by something else?
+            //if (response != IntPtr.Zero)
+            //{
+            //    var responseBerval = Marshal.PtrToStructure<NativeMethods.berval>(response);
+            //    if (responseBerval.bv_val != IntPtr.Zero)
+            //    {
+            //        Marshal.FreeHGlobal(responseBerval.bv_val);
+            //    }
+            //    Marshal.FreeHGlobal(response);
```

Comment:
I _think_ that the memory is cleaned up at the end of the PInvoke call since, which could potentially be causing the crash